src/Lkl.cc

```diff
@@ -661,16 +670,23 @@ Double_t Lkl::CallMinimization(Double_t g,Bool_t isVerbose,Int_t strategy)
 
       // try changing precision
       if(iflag!=0 || TMath::IsNaN(GetParErr(gGParIndex)))
-	fMinuit->DefineParameter(gGParIndex,fParName[gGParIndex],g,fParDelta[gGParIndex]/10.*TMath::Power(1.8,counter), 0, 0);
+        {
+          if(fParName[gGParIndex]=="tau")
+            fMinuit->DefineParameter(gGParIndex,fParName[gGParIndex],g,fParDelta[gGParIndex]/10.*TMath::Power(1.8,counter), 0, 10);
+          else
+            fMinuit->DefineParameter(gGParIndex,fParName[gGParIndex],g,fParDelta[gGParIndex]/10.*TMath::Power(1.8,counter), 0, 0);
+        }
       
       counter++;
-      if(counter==maxcounts)
+      if(iflag != 0 && counter==maxcounts)
```

Comment:
Taken

Reply:
Making a lot of testing I found some very edgy cases :-D the converged status but still failed really annoyed me when it meant loosing 24h ^^